plugins/gradle/java/src/service/project/JavaGradleProjectResolver.java

```diff
@@ -50,6 +50,9 @@
 @Order(ExternalSystemConstants.UNORDERED)
 public final class JavaGradleProjectResolver extends AbstractProjectResolverExtension {
 
+  private final IdentityHashMap<GradleBuildScriptClasspathModel, List<BuildScriptClasspathData.ClasspathEntry>> buildScriptEntriesMap =
```

Comment:
Storing state here will cause memory leak, as this class is a long-living service. It is not re-created on each sync.

Reply:
Looks better now, thank you!